docs/sdk/examples/assets/pastefromoffice/pastefromoffice.js

```diff
@@ -0,0 +1,164 @@
+var sources = [
+    {
+        name: 'Google Docs',
+        link: 'pastefromgoogledocs.html',
+        linkName: 'Paste from Google Docs',
+        regex: /id=(\"|\')?docs\-internal\-guid\-/,
+        isAlreadyShow: false
+    },
+    {
+        name: 'Microsoft Word',
+        link: 'pastefromword.html',
+        linkName: 'Paste from Word',
+        regex: getProperWordRegex(),
+        isAlreadyShow: false
+    },
+    {
+        name: 'Microsoft Excel',
+        link: 'pastefromexcel.html',
+        linkName: 'Paste from Excel',
+        regex: /<meta\s*name=generator\s*content="Microsoft\s*Excel\s*\d+"?\/?/gmi,
+        isAlreadyShow: false
+    },
+    {
+        name: 'LibreOffice',
+        link: 'pastefromlibreoffice.html',
+        linkName: 'Paste from LibreOffice',
+        regex: /<meta\s+name=["']?generator["']?\s+content="LibreOffice\s*\d+"?\/?/gmi,
+        isAlreadyShow: false
+    }
+];
+var instances = getInstances();
+// A state variable that indicates if the clipboard notification has been seen.
+// We use the variable for displaying the notification only once per demo.
+let hasNotificationBeenSeen = false;
+
+instances.forEach( function ( instanceName ) {
+    var editor = CKEDITOR.instances[ instanceName ];
+
+    if ( !editor ){
+        return;
+    }
+
+    editor.on( 'instanceReady', function() {
```

Comment:
There is a small issue with the code. If some instance has been initialized after adding the script to the HTML file, it won't attach events to this specific editor. Instead of looking into `CKEDITOR.instances` and listening on `instance[instanceReady]` event, you could attach the same event to the `CKEDITOR` namespace directly, see https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR.html#event-instanceReady
e.g.
```js
CKEDITOR.on( 'instanceReady', function( evt ) {
  var editor = evt.editor;
  editor.on( 'paste', ... );
} );
```